OptimizedRun fails for configuration containing boolean

### Issue description

I am trying to use micronaut-aot with the micronaut application I am developing as part of Micronaut Live. 

Steps to reproduce. 

Clone the repo: https://github.com/micronaut-advocacy/micronaut-live-newsletter
Checkout the branch `aot`
- Run postgreSQL via docker and export two environment variables as described here: 

https://github.com/micronaut-advocacy/micronaut-live-newsletter/blob/master/README.md#local-development

- Run ` ./gradlew optimizedRun` and you will get:
```
> Task :prepareJitOptimizations
11:28:08.814 [main] INFO  i.m.aot.MicronautAotOptimizer - Detected environments: []
11:28:09.015 [main] INFO  i.m.a.s.s.YamlPropertySourceGenerator - Converting application.yml into Java based configuration
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at io.micronaut.aot.cli.Main.executeInIsolatedLoader(Main.java:118)
        at io.micronaut.aot.cli.Main.run(Main.java:92)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at io.micronaut.aot.cli.Main.execute(Main.java:139)
        at io.micronaut.aot.cli.Main.main(Main.java:143)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.micronaut.aot.cli.Main.executeInIsolatedLoader(Main.java:112)
        ... 10 more
Caused by: java.lang.UnsupportedOperationException: Configuration map contains an entry of type class java.lang.Boolean which is not supported yet. Please file a bug report.
        at io.micronaut.aot.std.sourcegen.MapPropertySourceGenerator.generateMap(MapPropertySourceGenerator.java:68)
        at io.micronaut.aot.std.sourcegen.MapPropertySourceGenerator.generate(MapPropertySourceGenerator.java:84)
        at io.micronaut.aot.core.codegen.AbstractSingleClassFileGenerator.generate(AbstractSingleClassFileGenerator.java:35)
        at io.micronaut.aot.std.sourcegen.YamlPropertySourceGenerator.createMapProperty(YamlPropertySourceGenerator.java:75)
        at io.micronaut.aot.std.sourcegen.YamlPropertySourceGenerator.generate(YamlPropertySourceGenerator.java:58)
        at io.micronaut.aot.std.sourcegen.AbstractStaticServiceLoaderSourceGenerator.generate(AbstractStaticServiceLoaderSourceGenerator.java:90)
        at io.micronaut.aot.core.codegen.ApplicationContextConfigurerGenerator.generate(ApplicationContextConfigurerGenerator.java:59)
        at io.micronaut.aot.MicronautAotOptimizer$Runner.execute(MicronautAotOptimizer.java:318)
        at io.micronaut.aot.MicronautAotOptimizer.execute(MicronautAotOptimizer.java:191)
        ... 15 more

> Task :prepareJitOptimizations FAILED
```

This is my build config: 

```groovy
plugins {
...
    id("io.micronaut.application") version "3.1.0-M1"
    id("io.micronaut.aot") version "3.1.0-M1"
}

...
micronaut {
   ...
    aot {
        version = "1.0.0-M2"
        //configFile = file("aot.properties")
        sealEnvironment =  true
        optimizeServiceLoading = true
        optimizeClassLoading = true
        convertYamlToJava = true
        precomputeOperations = true
    }
}
```

--- a/aot-std-optimizers/src/main/java/io/micronaut/aot/std/sourcegen/MapPropertySourceGenerator.java
+++ b/aot-std-optimizers/src/main/java/io/micronaut/aot/std/sourcegen/MapPropertySourceGenerator.java
@@ -62,7 +62,7 @@ public class MapPropertySourceGenerator extends AbstractSingleClassFileGenerator
                 mapBuilder.add("null");
             } else if (CharSequence.class.isAssignableFrom(value.getClass())) {
                 mapBuilder.add("\"" + value + "\"");
-            } else if (Number.class.isAssignableFrom(value.getClass())) {
+            } else if (Number.class.isAssignableFrom(value.getClass()) || Boolean.class.isAssignableFrom(value.getClass())) {
                 mapBuilder.add(String.valueOf(value));
             } else {
                 throw new UnsupportedOperationException("Configuration map contains an entry of type " + value.getClass() + " which is not supported yet. Please file a bug report.");

--- a/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/MapPropertySourceGeneratorTest.groovy
+++ b/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/MapPropertySourceGeneratorTest.groovy
@@ -110,13 +110,17 @@ public class TestStaticPropertySource extends MapPropertySource {
         }
 
         where:
-        value      | literal
-        'a string' | '"a string"'
-        (int) 1    | '1'
-        (short) 1  | '1'
-        (byte) 1   | '1'
-        (long) 1   | '1'
-        2f         | '2.0'
-        2d         | '2.0'
+        value         | literal
+        'a string'    | '"a string"'
+        (int) 1       | '1'
+        (short) 1     | '1'
+        (byte) 1      | '1'
+        (long) 1      | '1'
+        2f            | '2.0'
+        2d            | '2.0'
+        true          | true
+        false         | false
+        Boolean.TRUE  | true
+        Boolean.FALSE | false
     }
 }

--- a/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy
+++ b/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy
@@ -46,6 +46,7 @@ public class Test_configStaticPropertySource extends MapPropertySource {
     super("test-config", new HashMap() {{
         put("micronaut.application.name", "demoApp");
         put("micronaut.server.port", 8181);
+        put("micronaut.server.cors.enabled", true);
         }});
   }
 }

--- a/aot-std-optimizers/src/test/resources/test-config.yml
+++ b/aot-std-optimizers/src/test/resources/test-config.yml
@@ -3,3 +3,5 @@ micronaut:
     name: demoApp
   server:
     port: 8181
+    cors:
+      enabled: true

